src/sample/Game.java

```diff
@@ -77,51 +59,19 @@ private boolean treasureIsFound()
         return this.bot.equals(this.treasure);
     }
 
-    /*private Task<Void> moveTask = new Task<>() {
-        @Override
-        protected Void call()
-        {
-            threadBot.MoveRandomly();
-            if(treasureIsFound())
-                completeGame();
-            return null;
-        }
-    };*/
-
-    // the below methods move the bot, check for treasure detection
-    // and then adjust the view
-    void moveUp()
+    void MoveInDirection(MoveEnum move)
     {
-        this.bot.Move(MoveEnum.Up);
-        treasureDetection();
-        Main.gameView.adjustBotAndTreasureLocations(this.bot, this.treasure);
-    }
-
-    void moveDown()
-    {
-        this.bot.Move(MoveEnum.Down);
-        treasureDetection();
-        Main.gameView.adjustBotAndTreasureLocations(this.bot, this.treasure);
-    }
-
-    void moveLeft()
-    {
-        this.bot.Move(MoveEnum.Left);
-        treasureDetection();
-        Main.gameView.adjustBotAndTreasureLocations(this.bot, this.treasure);
-    }
-
-    void moveRight()
-    {
-        this.bot.Move(MoveEnum.Right);
+        this.bot.Move(move);
         treasureDetection();
-        Main.gameView.adjustBotAndTreasureLocations(this.bot, this.treasure);
+        Main.gameView.redrawGrid(this.bot, this.treasure);
     }
 
     // treasure detection
     private void treasureDetection()
     {
-        if(treasureIsFound())
+        if(treasureIsFound()) {
+            Constants.GAME_COMPLETE = true;
```

Comment:
Setting a constant is counter intuitive, if you're modifying it, it's not constant.  Maybe make this a field on the game?

Reply:
Autoplay being implemented in Main would make this bit involved but I could make it public and proceed on that?

Reply:
Main is where the game is created, so you could do like ```Game.GameComplete``` in there, and then set that variable on the game here.